e2e/test_operator/operator_test.go

```diff
@@ -124,6 +126,265 @@ var _ = Describe("Operator", Label("e2e"), func() {
+							TimeAdded: &metav1.Time{Time: time.Now().Add(-time.Millisecond * time.Duration(taintKeyStarDuration*1000+int64(rand.Intn(1000))))},
+						},
+					}
+					log.Printf("Taint node: Pod name:%s Node name:%s Node taints:%+v TaintTime:%+v Now:%+v\n", pod.Name,
+						node.Name, node.Spec.Taints, node.Spec.Taints[0].TimeAdded.Time, time.Now())
+					fdbCluster.UpdateNode(node)
+				}
+
+				// Wait for operator to replace the pod
+				// time.Sleep(time.Second * time.Duration(*fdbCluster.GetAutomationOptions().Replacements.TaintReplacementTimeSeconds+1))
+				err := fdbCluster.WaitForReconciliation()
+				log.Printf("Reconciliation error:%+v\n", err)
+				Expect(err).NotTo(HaveOccurred())
+
+				for i, pod := range replacedPods {
+					log.Printf("Ensure %dth pod:%s is deleted\n", i, pod.Name)
+					fdbCluster.EnsurePodIsDeletedWithCustomTimeout(pod.Name, ensurePodIsDeletedTimeoutMinutes)
+				}
+			})
+		})
+
+		When("cluster disables taint feature with star taint key", func() {
+			taintKeyMaintenanceDuration = int64(1)
+			taintKeyStar = "*"
+			taintKeyStarDurationNegative := int64(-5)
+
+			BeforeEach(func() {
+				// Custom TaintReplacementOptions to taintKeyStar
+				fdbCluster.SetClusterTaintConfig([]fdbv1beta2.TaintReplacementOption{
+					{
+						Key:               &taintKeyStar,
+						DurationInSeconds: &taintKeyStarDurationNegative,
+					},
+				}, pointer.Int(1))
+
+				curClusterSpec := fdbCluster.GetCluster().Spec.DeepCopy()
+				Expect(len(curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions)).To(BeNumerically(">=", 1))
+				Expect(*curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions[0].Key).To(Equal(taintKeyStar))
+				Expect(*curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions[0].DurationInSeconds).To(BeNumerically("<", 0))
+
+				Expect(*fdbCluster.GetAutomationOptions().Replacements.Enabled).To(BeTrue())
+
+			})
+
+			It("should not remove any Pod whose Nodes are tainted", func() {
+				replacedPods := fixtures.RandomPickPod(initialPods.Items, numNodesTainted)
+				var replacedPodProcessGroupIDs []fdbv1beta2.ProcessGroupID
+				for _, pod := range replacedPods {
+					replacedPodProcessGroupIDs = append(replacedPodProcessGroupIDs, internal.GetProcessGroupIDFromMeta(fdbCluster.GetCluster(), pod.ObjectMeta))
+					// Taint replacePod's node
+					node := fdbCluster.GetNode(pod.Spec.NodeName)
+					node.Spec.Taints = []corev1.Taint{
+						{
+							Key:       taintKeyMaintenance,
+							Value:     "rack_maintenance",
+							Effect:    corev1.TaintEffectPreferNoSchedule,
+							TimeAdded: &metav1.Time{Time: time.Now().Add(-time.Millisecond * time.Duration(taintKeyStarDuration*1000+int64(rand.Intn(1000))))},
+						},
+					}
+					log.Printf("Taint node: Pod name:%s Node name:%s Node taints:%+v TaintTime:%+v Now:%+v\n", pod.Name,
+						node.Name, node.Spec.Taints, node.Spec.Taints[0].TimeAdded.Time, time.Now())
+					fdbCluster.UpdateNode(node)
```

Comment:
In this test case we are not resetting the node.

Reply:
It's in the AfterEach() in Context(). 
https://github.com/FoundationDB/fdb-kubernetes-operator/blob/05f282b84b022ec18d866ba30b1e5816a345e697/e2e/test_operator/operator_test.go#L166-L184